src/client/ua_client_subscriptions.c

```diff
@@ -1007,6 +1007,37 @@ UA_Client_MonitoredItems_modify_async(UA_Client *client,
     return statusCode;
 }
 
+static void *
+UA_MonitoredItem_findByID(void *data, UA_Client_MonitoredItem *mon) {
```

Comment:
I like to have `UA_MonitoredItem_findByID`.
But it should do the zip-tree-walking internally (or with another static method).
This is then also consistent with the other _findById methods. And it can be easily reused.